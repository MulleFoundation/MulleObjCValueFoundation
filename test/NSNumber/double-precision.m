#ifndef __MULLE_OBJC__
# import <Foundation/Foundation.h>
#else
# import <MulleObjCValueFoundation/MulleObjCValueFoundation.h>
#endif

#include <float.h>
#include <math.h>


static double values[] =
{
   0.8,
   8.2,
   8.3,
   11.7,
   11.07,
   11.007,
   11.0007,
   11.00007,
   11.000007,
   11.0000007,
   11.00000007,
   11.000000007,
   11.0000000007,
   11.00000000007,
   11.000000000007,
   11.0000000000007,
   11.00000000000007,
   11.000000000000007
};


#define sizeof_array( array) ( sizeof( array) / sizeof( array[ 0]))


int   main( void)
{
   NSNumber    *nr;
   NSUInteger  i;

   for( i = 0; i < sizeof_array( values); i++)
   {
      nr  = [NSNumber numberWithDouble:values[ i]];
      
      printf( "%%0.17g=%0.17g --- %%0.16g=%0.16g --- %%g=%g ::: description=%s\n", 
			values[ i], 
			values[ i], 
			values[ i], 
			[[nr description] UTF8String]);
   }
   return( 0);
}
