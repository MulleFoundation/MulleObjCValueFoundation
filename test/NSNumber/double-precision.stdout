%0.17g=0.80000000000000004 --- %0.16g=0.8 --- %g=0.8 ::: description=0.8
%0.17g=8.1999999999999993 --- %0.16g=8.199999999999999 --- %g=8.2 ::: description=8.199999999999999
%0.17g=8.3000000000000007 --- %0.16g=8.300000000000001 --- %g=8.3 ::: description=8.300000000000001
%0.17g=11.699999999999999 --- %0.16g=11.7 --- %g=11.7 ::: description=11.7
%0.17g=11.07 --- %0.16g=11.07 --- %g=11.07 ::: description=11.07
%0.17g=11.007 --- %0.16g=11.007 --- %g=11.007 ::: description=11.007
%0.17g=11.0007 --- %0.16g=11.0007 --- %g=11.0007 ::: description=11.0007
%0.17g=11.000069999999999 --- %0.16g=11.00007 --- %g=11.0001 ::: description=11.00007
%0.17g=11.000007 --- %0.16g=11.000007 --- %g=11 ::: description=11.000007
%0.17g=11.000000699999999 --- %0.16g=11.0000007 --- %g=11 ::: description=11.0000007
%0.17g=11.00000007 --- %0.16g=11.00000007 --- %g=11 ::: description=11.00000007
%0.17g=11.000000007000001 --- %0.16g=11.000000007 --- %g=11 ::: description=11.000000007
%0.17g=11.0000000007 --- %0.16g=11.0000000007 --- %g=11 ::: description=11.0000000007
%0.17g=11.000000000069999 --- %0.16g=11.00000000007 --- %g=11 ::: description=11.00000000007
%0.17g=11.000000000007001 --- %0.16g=11.000000000007 --- %g=11 ::: description=11.000000000007
%0.17g=11.0000000000007 --- %0.16g=11.0000000000007 --- %g=11 ::: description=11.0000000000007
%0.17g=11.000000000000069 --- %0.16g=11.00000000000007 --- %g=11 ::: description=11.00000000000007
%0.17g=11.000000000000007 --- %0.16g=11.00000000000001 --- %g=11 ::: description=11.00000000000001
